src/plugins/github/fetchGithubRepo.js

```diff
@@ -163,47 +162,56 @@ function tryGithubFetch(fetch, fetchOptions): Promise<any> {
             x.errors[0].type === "RATE_LIMITED"
           ) {
             return Promise.reject(
-              ({
-                type: "RATE_LIMIT_EXCEEDED",
-                retry: false,
-                error: x,
-              }: GithubResponseError)
+              ({type: "RATE_LIMIT_EXCEEDED", error: x}: GithubResponseError)
             );
           } else {
             return Promise.reject(
-              ({
-                type: "GRAPHQL_ERROR",
-                retry: false,
-                error: x,
-              }: GithubResponseError)
+              ({type: "GRAPHQL_ERROR", error: x}: GithubResponseError)
             );
           }
         }
         if (x.data === undefined) {
           if (x.message && x.message.includes("Bad credentials")) {
             return Promise.reject(
-              ({
-                type: "BAD_CREDENTIALS",
-                retry: false,
-                error: x,
-              }: GithubResponseError)
+              ({type: "BAD_CREDENTIALS", error: x}: GithubResponseError)
             );
           } else {
             // See https://github.com/sourcecred/sourcecred/issues/350
             return Promise.reject(
-              ({type: "NO_DATA", retry: true, error: x}: GithubResponseError)
+              ({type: "NO_DATA", error: x}: GithubResponseError)
             );
           }
         }
         return Promise.resolve(x.data);
       }),
     (e) =>
-      Promise.reject(
-        ({type: "FETCH_ERROR", retry: false, error: e}: GithubResponseError)
-      )
+      Promise.reject(({type: "FETCH_ERROR", error: e}: GithubResponseError))
   );
 }
 
+function errorDisposition(
+  e: GithubResponseError
+): AttemptOutcome<empty, GithubResponseError> {
+  if (
+    e.type === "FETCH_ERROR" ||
+    e.type === "GRAPHQL_ERROR" ||
+    e.type === "BAD_CREDENTIALS"
+  ) {
+    return {type: "FATAL", err: e};
+  }
+  if (e.type === "GITHUB_INTERNAL_EXECUTION_ERROR" || e.type === "NO_DATA") {
+    return {type: "RETRY", err: e};
+  }
+  if (e.type === "RATE_LIMIT_EXCEEDED") {
+    // Wait in 15-minute increments. TODO(@wchargin): Ask GitHub
+    // when our token resets (`{ rateLimit { resetAt } }`) and wait
+    // until just then.
+    const delayMs = 15 * 60 * 1000;
+    return {type: "WAIT", until: new Date(Date.now() + delayMs), err: e};
+  }
+  throw new Error((e.type: empty));
```

Comment:
This function is called in a catch block, so throwing here would cause an uncaught error. The previous implementation just returned a `{type: "FATAL", err: e}` instead, seems like that's what we would want here as well?

Reply:
This line is unreachable, which you can tell because it has access to a
value at the empty type. We only hit this case if the types are unsound.

This `throw new Error((...: empty))` pattern is very common throughout
our codebase; try `git grep `: empty` for examples.